### PR TITLE
rake data:validate

### PR DESCRIPTION
**Pull request by neo-tahi**
very basic reactive data integrity strategy

### CODE REVIEW

**Comment by Bestra** [comment]
Off the top of my head, in the past most of our 'bad data' problems on staging were related to a mismatch in task types and MMTs and creating new papers (it's been a while).  Would those records have been caught as 'invalid'?  I think the logic for this is fine, just want to make sure that we catch our most common scenarios as long as we're doing the work.

File: lib/tasks/data_load.rake [hunk 1]
```diff
@@ -174,4 +174,27 @@ namespace :data do
...
   end
+
+  task validate: :environment do
+    tables = ActiveRecord::Base.connection.tables;nil
+
+    tables.each do |table|
+      begin
+        z = table.singularize.camelize.constantize
+        records = z.all;nil
+
```
Comment (ryanstocker) [inline lib/tasks/data_load.rake]: I believe ```Table.all``` will load everything into memory.   ```Table.find_each do |record|``` loads 1,000 at a time - may be the way to go in the event of large tables.

**Comment by neo-tahi** [comment]
@Bestra - no, this script doesn't catch invalid Task types, but we can address that,
and @ryanstocker, good looking out.  1000+ records isn't something we're bumping into yet, but this pattern is more than reasonable, so we'll get it in there. Thanks!

File: lib/tasks/data_load.rake [hunk 2]
```diff
@@ -174,4 +174,27 @@ namespace :data do
...
+            i = i + 1
+            puts record.inspect
+          end
+        end;nil
+
+        p "====> #{i} Issues in #{z.to_s}"
+      rescue
+        p "error on #{table}... this table probably does not correspond to a Rails model"
+      end
+    end;nil
```
Comment (mikem) [inline lib/tasks/data_load.rake]: What's with the `;nil` in these couple of places? Is it a copy & paste artifact from when you tried this in IRB? I don't think it's necessary here.

**Comment by mikem** [comment]
Looks alright to me.

File: lib/tasks/data_load.rake (hunk 1 above)
Comment (afomi) [inline lib/tasks/data_load.rake]: hi @ryanstocker, 

`find_each` is now used in the lookup to batch queries.

**Pull request merged**